llmfoundry/common/fdiff.py

```diff
@@ -25,39 +25,38 @@ def __init__(self, diff_train_metrics=False, diff_eval_metrics=True):
 
     def batch_end(self, state: State, logger: Logger):
         if self.diff_train_metrics:
+            loss = state.loss.item()  # type: ignore
```

Comment:
I'm guessing the type error here is because `state.loss` can either be a tensor or a list. In this case, the better approach is to throw an error here that multiple losses are not supported by this callback. e.g. see: https://github.com/mosaicml/composer/blob/dev/composer/algorithms/mixup/mixup.py#L203